src/domain/usecases/MigrateUserNameUseCase.ts

```diff
@@ -0,0 +1,171 @@
+
+        if (options.csvPath) {
+            await this.generateCsvReport(options, usersToChange, fromAttribute, toAttribute);
+        }
+
+        if (options.post) {
+            const migrateResult = await this.userRepository.saveAll(
+                usersToChange,
+                fromAttribute,
+                toAttribute
+            );
+
+            if (options.sendNotification && !migrateResult.errorMessage) {
+                await this.sendNotifications(options, usersToChange, fromAttribute, toAttribute);
+            }
+
+            return migrateResult;
+        }
+
+        return {
+            errorMessage: "",
+            stats: {
+                ignored: 0,
+                updated: 0,
+            },
+        };
+    }
+
+    private parseToCsv(users: User[], from: UserAttribute, to: UserAttribute) {
+        return users.map(user => {
+            const { fromValue, toValue } = this.getValueFromProperties(user, from, to);
+            return {
+                id: user.id,
+                [from.key]: fromValue,
+                [to.key]: toValue,
+            };
+        });
+    }
+
+    private async generateCsvReport(
+        options: MigrateOptions,
+        users: User[],
+        from: UserAttribute,
+        to: UserAttribute
+    ) {
+        const csvWriter = createObjectCsvWriter({
+            path: options.csvPath,
+            header: [
+                {
+                    id: "id",
+                    title: "ID",
+                },
+                {
+                    id: options.from,
+                    title: options.from,
+                },
+                {
+                    id: options.to,
+                    title: options.to,
+                },
+            ],
+        });
+
+        logger.debug(`Generate report: ${options.csvPath}`);
+        await csvWriter.writeRecords(this.parseToCsv(users, from, to));
+    }
+
+    private async sendNotifications(
+        options: MigrateOptions,
+        users: User[],
+        from: UserAttribute,
+        to: UserAttribute
+    ) {
+        const emailContent = await readJson(options.emailPathTemplate);
+        const template = _.template(emailContent.body);
+        const attachments = (emailContent.attachments || []).map(
+            (path): Attachment => ({
+                type: "file",
+                path: path,
+            })
+        );
+
+        await promiseMap(users, user => {
+            const { fromValue, toValue } = this.getValueFromProperties(user, from, to);
+            return this.notificationsRepository.send({
+                recipients: [user.email],
+                bcc: options.adminEmail ? [options.adminEmail] : undefined,
+                subject: emailContent.subject,
+                body: {
+                    type: "html",
+                    contents: template({
+                        from: fromValue,
+                        to: toValue,
+                    }),
```

Comment:
I think that's wrong, in the user we need to notify the old/new values of the "to" property.